Userland/Libraries/LibWeb/Layout/Node.cpp

```diff
@@ -874,6 +874,15 @@ void NodeWithStyle::reset_table_box_computed_values_used_by_wrapper_to_init_valu
     mutable_computed_values.set_margin(CSS::InitialValues::margin());
 }
 
+void NodeWithStyle::set_fit_content_width_if_node_is_button_with_auto_width()
+{
+    // If the computed value of 'inline-size' is 'auto', then the used value is the fit-content inline size.
+    if (!(dom_node() && dom_node()->is_html_button_element() && m_computed_values.width().is_auto()))
+        return;
+    auto& mutable_computed_values = static_cast<CSS::MutableComputedValues&>(m_computed_values);
+    mutable_computed_values.set_width(CSS::Size::make_fit_content());
+}
```

Comment:
Why does this need to be a helper in `NodeWithStyle` instead of being something `TreeBuilder` can take care of?

Reply:
That's because we can't mutate computed values from `TreeBuilder`: `computed_values()` returns `ImmutableComputedValues` that can't be casted to `MutableComputedValues`.

Reply:
You can massage that into a `MutableComputedValues&` in a slightly disgusting way, like we do in many other places, e.g:

```cpp
auto& viewport_computed_values =
	const_cast<CSS::MutableComputedValues&>(
		static_cast<CSS::MutableComputedValues const&>(
			static_cast<CSS::ComputedValues const&>(
				viewport.computed_values()
			)
		)
	);
```

That said, perhaps having a `MutableComputedValues& mutable_computed_values()` accessor on `NodeWithStyle` would be nicer overall :sweat_smile:

Reply:
Yes, makes sense, I added `mutable_computed_values()` :)